core/src/main/scala/za/co/absa/spline/harvester/dispatcher/HttpLineageDispatcher.scala

```diff
@@ -36,24 +36,29 @@ import scala.util.control.NonFatal
 /**
  * HttpLineageDispatcher is responsible for sending the lineage data to spline gateway through producer API
  */
-class HttpLineageDispatcher(restClient: RestClient)
+class HttpLineageDispatcher(restClient: RestClient, config: HttpLineageDispatcherConfig)
```

Comment:
Since `restClient` object already incorporates some of the parameters specified in the `config`, passing both objects into the constructor creates an implicit ambiguity about the source of truth for those parameters. For example, from the constructor caller perspective, the `timeout` settings would appear as it has to be specified twice - in the dispatcher config, and the rest client. 
I would rather avoid passing config object together with other objects that are logically derived from it.